src/twisted/internet/defer.py

```diff
@@ -1169,6 +1178,10 @@ def __cmp__(self, other: object) -> int:
         return -1
 
 
+_DeferredListSingleResultT = Tuple[_DeferredResultT, int]
+_DeferredListResultItemT = Tuple[bool, _DeferredResultT]
+_DeferredListResultListT = List[_DeferredListResultItemT]
+
```

Comment:
These are used below, depending on `fireOnOnceCallback`, which make this API challenging to type well.

@mthuurne you might wanna double check me here, since you took the first pass.

Reply:
I'll dive into it when I'm a bit better rested.

Reply:
One thing to consider is whether these type aliases should be private or public. If they're being returned by public methods, they should probably be public.

While pydoctor currently doesn't document type aliases as separate entities, I think it is inevitable that we'll implement that at some point. After all, we do show type annotations in the docs now and the user will want to know what concrete types an alias represents. Also I think it should become possible in the future to have docstrings on aliases. To prepare for that future, it would make sense to make all type aliases that the user might want to learn about public.

Another reason to make type aliases public when returned by public methods is that the user might want to use those same aliases to annotate their own code.

Reply:
OK I was on the fence here, but will make them public.

Reply:
Actually, one reason I hesitated is that these are `TypeVar`s (via _DeferredResultT), and I'm not sure what the convention we want to use for those symbols would be.  Is it `DeferredListResultListT` or `DeferredListResultList`?  I think the former, because `TypeVar`s are different and I think name should reflect it.

@graingert You might have an opinion on this? I think I've seen you use `T_Foo` in examples, but I don't know if that's in use in Twisted. (I don't see it in Twisted proper.)

(We could also defer this to a separate PR.)

Reply:
Since `mypy` will tell you the expended type, and that's largely usable, I'm going to defer this for another PR, as I'm hesitant to add public symbols without more confidence that they are what we want.